-- Add the ALL option to the doc target and install HTML documentation.
-- Upstream's doc target no longer builds docs as part of the default build,
-- and it lacks an install rule. This patch restores both behaviours but
-- only installs the generated HTML output, since Doxygen XML output is not
-- generated by default.

--- doc/CMakeLists.txt.orig	2026-06-28 23:49:10 UTC
+++ doc/CMakeLists.txt
@@ -69,9 +69,15 @@ configure_file(Doxyfile.in Doxyfile @ONLY)
 configure_file(Doxyfile.in Doxyfile @ONLY)
 
 # note the option ALL which allows to build the docs together with the application
-add_custom_target(doc
+add_custom_target(doc ALL
   COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
   COMMENT "Generating API documentation with Doxygen"
   VERBATIM
   )
+
+if(FTXUI_ENABLE_INSTALL)
+  install(DIRECTORY ${PROJECT_BINARY_DIR}/doc/doxygen/html
+          DESTINATION ${CMAKE_INSTALL_DOCDIR}
+          COMPONENT doc)
+endif()
